mantidproject/mantid
# D2B detector alignment

This PR implements additional features requested for D2B (see the issue).

**To test:**
This script effectively exercises all the new features. It should run and produce a 2D result (inside ws group) that looks like the attachment. Files are attached in a tarball; chop off `.zip`.
Quantitatively it is tested with the unit and system tests.

```
CALIBFILE = '/Users/vardanyan/Desktop/test-calibPR.nxs' # give some path on your machine
calib = PowderDiffILLDetEffCorr(CalibrationRun='532008,532009',
                                                        DerivationMethod='GlobalSummedReference2D',
                                                        ExcludedRange=[-5,10],
                                                        OutputWorkspace='calib',
                                                        MaskCriterion=[0.3,3])
calibmax = mtd['calib'].extractY().flatten().max()
assert(calibmax <= 3)
SaveNexusProcessed(InputWorkspace='calib', Filename=CALIBFILE)
ws = PowderDiffILLDetScanReduction(Run = '535401,535402', Output2D=True, Output1D=False, CalibrationFile=CALIBFILE)
```

![screen shot 2018-04-24 at 10 26 29](https://user-images.githubusercontent.com/16865759/39175467-b2dc9c0c-47aa-11e8-938c-cd353a633655.png)

[files4PR.tar.gz.zip](https://github.com/mantidproject/mantid/files/1941715/files4PR.tar.gz.zip)



Also merge [PR#110](https://github.com/mantidproject/mantidgeometry/pull/110).
Fixes #22274 and #22355. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
